ts/src/gate.ts

```diff
@@ -3349,9 +3353,6 @@ export default class gate extends Exchange {
                         'expiration': expiration, // required, how long (in seconds) to wait for the condition to be triggered before cancelling the order
                     };
                 }
-                if (timeInForce !== undefined) {
```

Comment:
I have not removed it, just moved in upper part mistakenly (reverted back).

Reply:
I mean, it's still in upper part of the code if you look at a bit above.

Reply:
oh ok my bad 😅

Reply:
@ttodua wait apparently the condition is different now (=== instead of !==)

Reply:
yes, it's changed, because the logic didnt seem consistent to me. to understand it, you should view the full body of code of previous version vs current version. 
let me go in a bit more detail to explain changes:

Reply:
in previous version, lines 3249 and 3328 were not correctly set imo, they shouldn't have been hardcoded, so i've changed that. 
now, we parse PO & TIF better in this implementation,  please take a look again.